src/server/scripts/Spells/spell_item.cpp

```diff
@@ -318,7 +318,7 @@ class spell_item_with_mount_speed : public SpellScriptLoader
         void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
         {
             Unit* target = GetTarget();
-            if (uint16 spellId = getMountSpellId())
+            if (uint32 spellId = getMountSpellId())
```

Comment:
you should put here and in the other one above `auto` and change instead in line 291 the return type uin16 to uint32 of the function `getMountSpellId()`

Reply:
Too long I wrote this, completely forgot about it